lib/realtime/server.ex

```diff
@@ -342,12 +378,17 @@ defmodule Realtime.Server do
     )
   end
 
-  @spec broadcast(t()) :: :ok
-  defp broadcast(state) do
+  @spec broadcast(t(), :vehicles | :alerts) :: :ok
+  defp broadcast(state, data_type) do
     registry_key = self()
 
     Registry.dispatch(Realtime.Supervisor.registry_name(), registry_key, fn entries ->
-      Enum.each(entries, &send_data(&1, state))
+      Enum.each(entries, fn {pid, subscripition_key} ->
+        if (data_type == :alerts and match?({:alerts, _}, subscripition_key)) or
+             (data_type == :vehicles and !match?({:alerts, _}, subscripition_key)) do
+          send_data({pid, subscripition_key}, state)
+        end
+      end)
```

Comment:
> Finally, there was a bit of a question as to whether to use Realtime.Server (like most of our real-time bus data) or the Phoenix PubSub (like the live train data).

suggestion (non-blocking): I don't have a  have a strong understanding of the technical differences between using `server` or `PubSub` yet so this may be a fully uniformed thought, but I do tend to like how the train & status data are managed separately from this server. It might be nice to have one file focused only on alerts data, whether that uses `PubSub` or is structured in the same way as `server`, so that this kind of conditional check isn't needed.

Reply:
I was a bit confused by the filter here, personally, it took a while to make sense of it due to indentation.

I agree with Kayla that removing this conditional and making it it's own codepath would possibly result in cleaner logic. I'd otherwise vote for filtering `entries` before using `Enum.each` instead of `if`s within the each lambda.

---

(Also I'm commenting now because I want the slack github reminders to stop on vacation lol)

Reply:
So my thinking here is that we might actually want to use this as a jumping off point to do more filtering in the `Realtime.Server` module around subscriptions. Currently, any new vehicle data triggers all vehicles to be pushed to their respective channels, even if those vehicles weren't actually updated, so I saw it as a bonus that this was creating logic to only updates to certain subscribers. I can create a tech debt ticket to actually do that filtering.

Reply:
Okay, gotcha. Yeah, I think my idea of only broadcasting for updated vehicles / alerts might be harder than I initially thought with the code as it is. I still think it's a worthy goal, though, since it might make it possible to cut down on how much data we need to push over websockets by a fair bit. It sounds like we basically need a rethink / rationalization of how realtime data like this is handled on the back-end. I'll write up some thoughts in a tech cleanup ticket.

Reply:
Asana ticket: https://app.asana.com/0/1152340551558956/1203441390849641/f